### ISSUE TITLE
y_flip not performed on user-supplied mip levels

### ISSUE THREAD

**Issue by pezcode** [BinomialLLC/basis_universal#257]
If `m_y_flip` is set, the encoder calls `flip_y()` on the main image, but not for any of the mip levels in `m_source_mipmap_images`.

Is this a bug?

Slightly related issue: This code checks for alpha before swizzling: https://github.com/BinomialLLC/basis_universal/blob/d7e2d4787986eed7173f763d9cf67c57928c6743/encoder/basisu_comp.cpp#L600
And the comment indicates that mip images are assumed to be pre-swizzled, but then they actually get swizzled again later.

**Comment by richgel999** [comment]
Yes, this is a bug. The m_source_mipmap_images feature was added as a PR and this fell through the cracks. Will fix.

Same for the swizzling feature.
